packages/web-api-scan-runner/src/runner/runner.ts

```diff
@@ -0,0 +1,88 @@
+    OnDemandScanResult,
+    ReportFormat,
+    ScanState,
+} from 'storage-documents';
+import { ScanMetadataConfig } from '../scan-metadata-config';
+import { ScannerTask } from '../tasks/scanner-task';
+import { WebDriverTask } from '../tasks/web-driver-task';
+
+@injectable()
+export class Runner {
+    constructor(
+        @inject(ScanMetadataConfig) private readonly scanMetadataConfig: ScanMetadataConfig,
+        @inject(Logger) private readonly logger: Logger,
+        @inject(ScannerTask) private readonly scannerTask: ScannerTask,
+        @inject(OnDemandPageScanRunResultProvider) private readonly onDemandPageScanRunResultProvider: OnDemandPageScanRunResultProvider,
+        @inject(WebDriverTask) private readonly webDriverTask: WebDriverTask,
+    ) {}
+
+    public async run(): Promise<void> {
+        const scanMetadata = this.scanMetadataConfig.getConfig();
+        let onDemandScanResult: OnDemandScanResult;
+        let onDemandPageScanReport: OnDemandPageScanReport;
+        let onDemandPageScanRunResult: OnDemandPageScanRunResult;
+
+        try {
+            const pageScanRunResult = (await this.onDemandPageScanRunResultProvider.readScanRuns([scanMetadata.id]))[0];
+
+            onDemandScanResult = {
+                state: 'unknown' as ScanState,
+                issueCount: 0,
+            };
+
+            onDemandPageScanReport = {
+                reportId: '',
+                format: 'sarif' as ReportFormat,
+                href: '',
+            };
+
+            // set scanned page run state to running
+            onDemandPageScanRunResult = {
+                state: 'running' as OnDemandPageScanRunState,
+                timestamp: new Date()
+                    .toJSON()
+                    .valueOf()
+                    .toString(),
+                error: '',
+            };
+
+            pageScanRunResult.run = onDemandPageScanRunResult;
+
+            await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanRunResult);
+
+            // start new web driver process
+            await this.webDriverTask.launch();
+
+            // scan page for accessibility issues
+            const axeScanResults: AxeScanResults = await this.scannerTask.scan(scanMetadata.url);
+
+            onDemandPageScanRunResult.state = 'completed' as OnDemandPageScanRunState;
+            onDemandPageScanRunResult.timestamp = new Date()
+                .toJSON()
+                .valueOf()
+                .toString();
+
+            onDemandScanResult.issueCount = axeScanResults.results.violations.length;
+            onDemandScanResult.state = (onDemandScanResult.issueCount > 0 ? 'fail' : 'pass') as ScanState;
+
+            pageScanRunResult.run = onDemandPageScanRunResult;
+            pageScanRunResult.reports = [onDemandPageScanReport];
+            pageScanRunResult.scanResult = onDemandScanResult;
+
+            await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanRunResult);
```

Comment:
am not sure if this will work. etag value will be different now, since you have already updated this object with a etag value. if you modify the object & call update api, since you are still having the old etag value it may not work.

Reply:
As discussed offline, it should be fine